src/components/messages-view-configurator/EditableRule.tsx

```diff
@@ -0,0 +1,122 @@
+/** *****************************************************************************
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************** */
+
+import React, { useState, useCallback } from 'react';
+import { useMessageDisplayRulesStore } from '../../hooks';
+import { MessageDisplayRule, MessageViewType } from '../../models/EventMessage';
+import SessionEditor from './SessionEditor';
+import RuleEditor from './RuleEditor';
+import Reorder from './Reorder';
+
+type EditableRuleProps = {
+	rule: MessageDisplayRule;
+	sessions: string[];
+	index: number;
+	isFirst?: boolean;
+	isLast?: boolean;
+	autofocus?: boolean;
+};
+
+const EditableRule = ({ sessions, rule, isFirst, isLast, index, autofocus }: EditableRuleProps) => {
+	const rulesStore = useMessageDisplayRulesStore();
+	const [session, setSession] = useState(rule.session);
+	const [viewType, setViewType] = useState<MessageViewType>(rule.viewType);
+	const [sessionIsEditing, setSessionIsEditing] = useState(false);
+	const [ruleIsEditing, setRuleIsEditing] = useState(false);
+
+	const deleteHandler = () => {
+		rulesStore.deleteMessagesDisplayRule(rule);
+	};
+
+	const editRuleSession = useCallback(() => {
+		rulesStore.editMessageDisplayRule(rule, { ...rule, session });
+		setSessionIsEditing(false);
+	}, [session]);
+
+	const editViewType = (vType: MessageViewType) => {
+		const newRule = { ...rule, viewType: vType };
+		if (rule.session === '*') {
+			rulesStore.setRootDisplayRule(newRule);
+			setRuleIsEditing(false);
+			return;
+		}
+		rulesStore.editMessageDisplayRule(rule, newRule);
+		setRuleIsEditing(false);
+	};
+
+	const renderSession = () => {
+		return sessionIsEditing && rule.editableSession ? (
```

Comment:
lets refactor these render functions to components inside this file Session and ViewType